### ISSUE TITLE
No strong name signed assemplies

### ISSUE THREAD

**Issue by Kultman** [jamesmontemagno/monkey-cache#99]
Trying to use Monkey Cache for a ClickOnce deployed windows app. I keep getting an error about assemblies not being signed. I've tried using .NET assembly Strong name-signer, which works for the Monkey cache dll and Monkey Cache sqllite dll, but then the problem just moves to the sqllite assemblies. Anyone else seen that problem?

**Comment by jamesmontemagno** [comment]
Have you looked at -> https://github.com/dsplaisted/strongnamer

**Comment by Kultman** [comment]
I found something like it, just installed Nuget Package and Bob became my uncle, so to speak... :) Brutal.Dev.StrongNameSigner package takes care of it all automatically, pretty nice.

But now I'm getting  error in stead... This error is for FileStore, but have ried with SqlLite also, same issue...
![image](https://user-images.githubusercontent.com/18451163/111743404-f52a0180-8889-11eb-88aa-6a04728695c0.png)

Error text is in danish, which is a pain, don't know how to get that to be in english :) Something about not being able to access a method to Get BasePath...

Thought it might be caused by rights to file system, my PC recently got put in our domain and I lost my local admin rights. So I thought moving Base Path to MyDocuments might help... but no...

        'BarrelUtils.SetBaseCachePath(Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments))
        Barrel.ApplicationId = "DigitalTryk"

I'm using MonkeyCache in a Xamarin project and have grown very fond of it, not being able to use it in a Windows app is a bit of a bummer. And yes, I code VB.NET for windows apps and C# for Xamarin and WebAPI - I'm originally from the VB comunity but am a bit agnostic about it now... :)

**Comment by jamesmontemagno** [comment]
What type of app is this that is throwing the error?

**Comment by jamesmontemagno** [comment]
You will probably need to set the cache directory directly if it isn't a platform I support -> https://github.com/jamesmontemagno/monkey-cache#persisting-data-longer

**Comment by Kultman** [comment]
It's a standard Winforms application in VB.NET, using Telerik controls. And I am setting cache directory to MyDocuments with BarrelUtils.SetBaseCachePath(Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments))

**Comment by jamesmontemagno** [comment]
I will give this a test to see. Is it .NET Core or Framework?

**Comment by Kultman** [comment]
It's a Framework 4.8 project

**Comment by jamesmontemagno** [comment]
So, I created a winforms app with framework 4.8 and it seems to be working alright here.
[WindowsFormsApp1.zip](https://github.com/jamesmontemagno/monkey-cache/files/6198810/WindowsFormsApp1.zip)

**Comment by Kultman** [comment]
Ya, I see that your project works. Mine still doesn't...

I'll been testing things to isolate problem. So far I've tested elevated admin rights, same problem. Created VB.NET project to see if it a VB specific problem, my clean VB.NET projects also works, so no it's not VB. Did a complete reinstall of all Nuget packages, did no difference. Turned off signing with no effect...

There is something in my projects that MonkeyCache really doesn't like tho... Still don't know what. But I'll create a new branch and peel off everything layer by layer and find out what is conflicting with Monkeycache. Don't have time for this within the next 2 weeks, so I'll get back to you on this...

**Comment by jamesmontemagno** [comment]
Could always grab the source code directly and add it to your project :) it is pretty much a single file :)

**Comment by Kultman** [comment]
I think I found the cause of this error. Was working on an old project and wanted to replace a SQLServerLocalDB implementation with MonkeyCache. Again I'm deploying with ClickOnce and have signed the assembly with a strong name key file. Have also sogned the manifest with a Code signing certificate.
The assembly signing gives an error, because your package is unsigned, so I installed Brutal.Dev.StrongNameSigner - this automatically signs unsigned packages. And this Nuget package gives me this mysterious error I reported. Removed assembly signing and kept the Code signing and now MonkeyCache seems to work.
